pkg/router/template/router.go

```diff
@@ -873,72 +873,103 @@ type endpointsCounter func(key string) int32
 // getServiceUnits returns a map of service keys to their weights.
 // The requests are loadbalanced among the services referenced by the route.
 // The weight (0-256, default 1) sets the relative proportions each
-// service gets (weight/sum_of_weights fraction of the requests).
+// service gets (weight/sum_of_weights) fraction of the requests.
+// Default when service weight is omitted is 1.
 // When the weight is 0 no traffic goes to the service. If they are
 // all 0 the request is returned with 503 response.
 // For each service, the requests are distributed among the endpoints.
 // Each endpoint gets weight/numberOfEndpoints portion of the requests.
+// The largest weight per endpoint is scaled to 256 to permit better
+// percision results.
+// returns weight per endpoint for each service
 // The above assumes roundRobin scheduling.
 func getServiceUnits(counter endpointsCounter, route *routeapi.Route) map[string]int32 {
+	type svcInfo struct {
+		weight   int32   // specified (desired) weight (0 - 256)
+		numEp    int32   // number of service endpoints
+		epWeight float32 // calculated weight per endpoint (0.0 - 256.0)
+	}
+	infoData := make(map[string]svcInfo)
+	var info svcInfo 
 	serviceUnits := make(map[string]int32)
-	key := endpointsKeyFromParts(route.Namespace, route.Spec.To.Name)
 
-	// find the maximum weight
-	var maxWeight int32 = 1
-	if route.Spec.To.Weight != nil && *route.Spec.To.Weight > maxWeight {
-		maxWeight = *route.Spec.To.Weight
+	// get the weight and number of endpoints for each service
+	key := fmt.Sprintf("%s/%s", route.Namespace, route.Spec.To.Name)
```

Comment:
isn't there a function for this?

Reply:
yes, we should use endpointsKeyFromParts()

Reply:
@pravisankar Thanks, changed.